### DISCUSSION TITLE
Failed to install hererocks

### DISCUSSION THREAD

**Discussion by anurag3301** [wbthomason/packer.nvim#937] [Q&A]
Packer is not able to install the rocks pacages. I have following in my plugins.lua
```lua
use {
    'anurag3301/nvim-platformio.lua',
    rocks = {'lunajson', 'luasec'},
    requires = {
        {'akinsho/nvim-toggleterm.lua'},
        {'nvim-telescope/telescope.nvim'},
        {'nvim-lua/plenary.nvim'}
    }
}
```
Neovim version: NVIM v0.7.0
Lua version: Lua 5.4.4 
Luarocks version: LuaRocks 3.9.0

I run `:PackerSync` and it doesn't install the rock pacages, I get following error when I check `:message`
```
[packer.nvim] [ERROR 19:32:52] luarocks.lua:104: Failed to install hererocks: {
[packer.nvim]   exit_code = 1,
[packer.nvim]   signal = 0
[packer.nvim] }
```
I tried install `hererocks` manually with `pip install hererocks` but no change, I still get the above mentioned error
Please help me how to deal with this

**Comment by jokesper** [comment]
I got the same issue. In my case it was because I didn't have `unzip` installed (Arch). Maybe try to get the error message of the command run in line 86-92 of luarocks.lua. You could get it by temporarly adding a `print(luarocks_cmd)` below line 92 in luarocks.lua. And getting it via `:message`. In my case it was `python /home/<your_user>/.cache/nvim/packer_hererocks/hererocks.py --verbose -j 2.1.0-beta3 -r latest /home/<your_user>/.cache/nvim/packer_hererocks/2.1.0-beta3` and my output was:
```
LuaJIT 2.1.0-beta3 already installed
Fetching LuaRocks 3.9.1 (cached)
Verifying SHA256 checksum
Building LuaRocks 3.9.1
Running ./configure --prefix=/home/<your_user>/.cache/nvim/packer_hererocks/2.1.0-beta3 --with-lua=/home/<your_user>/.cache/nvim/packer_hererocks/2.1.0-beta3

Configuring LuaRocks version 3.9.1...

Lua version detected: 5.1
Lua interpreter found: /home/<your_user>/.cache/nvim/packer_hererocks/2.1.0-beta3/bin/lua
lua.h found: /home/<your_user>/.cache/nvim/packer_hererocks/2.1.0-beta3/include/lua.h
Could not find 'unzip'.
Make sure it is installed and available in your PATH.

configure failed.

Error: got exitcode 1 from command ./configure --prefix=/home/<your_user>/.cache/nvim/packer_hererocks/2.1.0-beta3 --with-lua=/home/<your_user>/.cache/nvim/packer_hererocks/2.1.0-beta3
```
After running it manually you could try to fix the error yourself or post it here.

**Reply by S1M0N38** [reply]
Yeah, that's a good strategy for debugging installation script. I'm on macOS 12.6 and turns out that there were two problem:
1. No more `python` executable (only `python3`) so I use [pyenv](https://github.com/pyenv/pyenv) to circumvent the problem.
2. During the compilation of LuaJIT I have to specify macOS version using an env var (`MACOSX_DEPLOYMENT_TARGET="12.6"`) (see [this](https://github.com/LuaJIT/LuaJIT/issues/764) issue for details)
